### ISSUE TITLE
[Bug]:  Accessibility issue while using @fluentui/react/lib/GroupedList

### ISSUE THREAD

**Issue by AdityaShankarMishra** [microsoft/fluentui#31290]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
We are currently using
@fluentui/react/lib/GroupedList, which includes an expand-collapse button in the group header, as depicted in the screenshot below. However, we've encountered an issue where the
tabIndex
property for the expand-collapse button is automatically set to -1. This is causing the following accessibility issue during keyboard navigation. 

"By invoking enter/space/alt + down/arrows keys Expand/collapse button is not accessible due to which table present under "Protected Item details" is not displayed".  
 
However in narrator when scan mode is on, it is accessible with down arrow key. As I think the "down-arrow" key is compensating for the tabindex="-1" we are seeing in the DOM. However with NVDA it is not working.
```

### Are you reporting Accessibility issue?

Yes

## Actual Behavior
Expand/collapse button present for 'Protected item details" tree grid is not keyboard accessible.
###Observation:

- By invoking enter/space/alt + down/arrows keys Expand/collapse button is not accessible due to which table present under "Protected Item details" is not displayed.
- However, the Expand/collapse button is accessible on mouse click.

## Expected Behavior
Expand/collapse button present for 'Protected item details" tree grid should be keyboard accessible.
![image](https://github.com/microsoft/fluentui/assets/136810064/365925df-a5dc-43b7-8c5f-855a6b1bf00a)

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by AdityaShankarMishra** [comment]
Hi @smhigley /@Hotell, any updates on this one?

**Comment by smhigley** [comment]
@AdityaShankarMishra thanks for logging an issue! The GroupedList should allow sections to expand/collapse by pressing left and right arrow keys while focus is on the row, not on that button. This works for me on our docs site: https://developer.microsoft.com/en-us/fluentui#/controls/web/groupedlist

The button is there so that screen reader users who are not using our built-in keyboard navigation can still expand and collapse sections. This includes people using windows screen readers in virtual cursor mode, VoiceOver on macOS, VoiceOver on iOS, or Talkback. It also enables access for voice control users.

That button does not need to be keyboard-navigable, since people navigating using the built-in keyboard commands would use the right/left arrow key behavior on the row. Allowing people to arrow or tab to the button would break expected treegrid navigation without providing any benefit.

I hope that answers your question! If the right/left arrow key expand/collapse behavior is not working in your product, please provide an isolated reproduction using the "Export to Codepen" button on our docs site. I can't access the example in your product, since I don't have any items in that table.

**Comment by AdityaShankarMishra** [comment]
Hi Sarah, thanks for your quick response, so in our case, after scan mode is on, once we reach to the row using TAB key, if we press right arrow key with an "enter" it is working. Based on our discussion offline, I am closing this one as this is a known behavior.
